common.php

```diff
@@ -265,19 +265,22 @@ function deregister_globals()
 	$config['plugins'][$k]['enabled'] = !empty($plugin['plugin_enabled']) ? true : false;
 	$config['plugins'][$k]['dir'] = !empty($plugin['plugin_dir']) ? ($plugin['plugin_dir'] . '/') : '';
 	// Plugins autoload - BEGIN
+	$plugin_dir = IP_ROOT_PATH . PLUGINS_PATH . $config['plugins'][$k]['dir'];
 	foreach ($class_plugins->plugin_includes_array as $plugin_include)
 	{
 		$config['plugins'][$k][$plugin_include] = !empty($plugin['plugin_' . $plugin_include]) ? true : false;
 		if (!empty($config['plugins'][$k][$plugin_include]))
 		{
-			@include_once(IP_ROOT_PATH . PLUGINS_PATH . $config['plugins'][$k]['dir'] . $plugin_include . '.' . PHP_EXT);
+			@include_once($plugin_dir . $plugin_include . '.' . PHP_EXT);
 		}
 	}
 
 	// if the plugin has a class (events, etc), register it.
-	$plugin_class_name = 'plugin_' . $k;
-	if (class_exists($plugin_class_name))
+	$plugin_class_name = 'class_plugin_' . $k;
```

Comment:
Why did you change the name of the class? I can't remember why I have used that name, but maybe some private plugins are using the other format... I need to check before integrating your code.

Reply:
After having looked at the code and analyzed other plugins, I still prefer to keep the inclusion of the autoload class in a simpler way, so every plugin may have a file called "class.php" within his root folder which gets included automatically. I will accept the name change of the class, but unless there is another good reason, I won't add another INCLUDE to common.php (it just complicate the code, without adding anything really beneficial, at least at a first look).

Reply:
`class.php` doesn't get loaded automatically, does it?